src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1418,8 +1415,13 @@ private void ReduceMemoryUsage(StorageEnvironment environment)
                 _currentMaximumAllowedMemory = DefaultMaximumMemoryAllocation;
 
                 var afterFree = NativeMemory.CurrentThreadStats.TotalAllocated;
-                if (_logger.IsInfoEnabled)
-                    _logger.Info($"After cleanup, using {new Size(afterFree, SizeUnit.Bytes)} by '{Name}'.");
+
```

Comment:
Do we want to check the previous `afterFree` with the current `beforeFree`, to skip all of those `Clean` methods?

Reply:
Not sure that I follow? You can get `afterFree` only you execute them.

Reply:
From the previous run

Reply:
cc @grisha-kotler ?

Reply:
@karmeli87 good idea.
https://github.com/ravendb/ravendb/pull/9781